TestFoundation/TestNSURLSession.swift

```diff
@@ -409,6 +410,32 @@ class TestURLSession : XCTestCase {
 
         waitForExpectations(timeout: 30)
     }
+
+    func test_timeoutInterval() {
+        let serverReady = ServerSemaphore()
+        globalDispatchQueue.async {
+            do {
+                try self.runServer(with: serverReady, startDelay: 3, sendDelay: 5, bodyChunks: 3)
+            } catch {
+                XCTAssertTrue(true)
+                return
+            }
+        }
+        serverReady.wait()
+        let config = URLSessionConfiguration.default
+        config.timeoutIntervalForRequest = 10
+        let session = URLSession(configuration: config, delegate: nil, delegateQueue: nil)
+        var expect = expectation(description: "download task with handler")
+        var req = URLRequest(url: URL(string: "http://127.0.0.1:\(serverPort)/Peru")!)
+        req.timeoutInterval = 1
+        var task = session.dataTask(with: req) { (data, _, error) -> Void in
+            defer { expect.fulfill() }
+            XCTAssertNotNil(error)
```

Comment:
I am expecting the error to be thrown in the passing scenario as the ```timeouInterval``` set for the request  is less than the ```sendDelay```  and hence am not printing the error.